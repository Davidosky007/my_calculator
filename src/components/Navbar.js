/* eslint-disable linebreak-style */
import React from 'react';
import { Link } from 'react-router-dom';

const Navbar = () => (
  <header className="main_nav">
    <div>
      <h1>Math Magician</h1>
    </div>
    <nav>
      <ul className="nav_list">
        <li>
          <Link to="/">
            Home
          </Link>
        </li>
        <li>
          <Link to="/calculator">
            Calculator
          </Link>
        </li>
        <li>
          <Link to="/qoutes">
            Qoutes
          </Link>
        </li>
      </ul>
    </nav>
  </header>
);

export default Navbar;
