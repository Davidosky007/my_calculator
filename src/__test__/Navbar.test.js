/* eslint-disable linebreak-style */
import { fireEvent, render, screen } from '@testing-library/react';
import { create } from 'react-test-renderer';
import {
  BrowserRouter,
} from 'react-router-dom';
import '@testing-library/jest-dom';
import Navbar from '../components/Navbar';

describe('test for Navbar components', () => {
  test('should navigate to home link', () => {
    const { getByText } = render(<a href="/">Home</a>);
    const link = getByText('Home');
    fireEvent.click(link);
    expect(screen.getByText('Home').closest('a')).toHaveAttribute('href', '/');
  });
  test('should navigate to calculate link', () => {
    const { getByText } = render(<a href="/calculator">Calculator</a>);
    const link = getByText('Calculator');
    fireEvent.click(link);
    expect(screen.getByText('Calculator').closest('a')).toHaveAttribute('href', '/calculator');
  });
  test('should navigate to calculate link', () => {
    const { getByText } = render(<a href="/qoutes">Qoutes</a>);
    const link = getByText('Qoutes');
    fireEvent.click(link);
    expect(screen.getByText('Qoutes').closest('a')).toHaveAttribute('href', '/qoutes');
  });
  test('Navbar component should render properly', () => {
    const navbar = create(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>,
    ).toJSON();
    expect(navbar).toMatchSnapshot();
  });
});
