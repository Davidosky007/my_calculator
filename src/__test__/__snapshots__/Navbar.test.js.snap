// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test for Navbar components Navbar component should render properly 1`] = `
<header
  className="main_nav"
>
  <div>
    <h1>
      Math Magician
    </h1>
  </div>
  <nav>
    <ul
      className="nav_list"
    >
      <li>
        <a
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li>
        <a
          href="/calculator"
          onClick={[Function]}
        >
          Calculator
        </a>
      </li>
      <li>
        <a
          href="/qoutes"
          onClick={[Function]}
        >
          Qoutes
        </a>
      </li>
    </ul>
  </nav>
</header>
`;
